### PR TITLE
Communicator docs

### LINKED ISSUE #8150: WinRM driver fails to connect
#### Overview of the Issue

WinRM fails to connect and packer build times out.

#### Reproduction Steps

`packer build -ver-file ${vars} ${config}`

I attempt with and without WinRM security. It seems that WinRM and PowerShell remoting expects at least self-signed certificates and allows you to ignore them. By hand I could get no other configuration but that to work.

It pains me to open this issue, but I can not find any usable documentation on this setup anywhere else. It is entirely possible I am doing something wrong, but I just fixed WinRM so that I can access the guest by hand.

### Packer version

1.4.3.

### Simplified Packer Buildfile
``` json
{
	"name": "windows_2012r2_base",
	
	"cpus": "2",
	"memory": "4096",
    "disk_size": "262144",
	
    "install_media": "",
    "install_media_checksum_type": "sha512",
    "install_media_checksum": "",
	"answer_media": "./media/windows_2012r2_base_setup.iso",
	"scripts_dir": "./media/windows_2012r2_base_setup",
	
	"output_directory": ".\\output\\win2012r2\\"
}
```

``` json
{
    "variables": {
		"name": null,
        "cpus": null,
        "memory": null,
        "disk_size": null,
        "install_media": null,
		"install_media_checksum_type": "file",
		"install_media_checksum": "{{ user `install_media` }}.sum",
		"answer_media": null,
		"scripts_dir": null,
		"output_directory": ""
    },
    "builders": [
    {
        "vm_name": "{{ user `name` }}",
        "type": "hyperv-iso",
        "output_directory": "{{ user `output_directory` }}",

        "iso_url": "{{ user `install_media` }}",
        "iso_checksum_type": "{{ user `install_media_checksum_type` }}",
        "iso_checksum": "{{ user `install_media_checksum` }}",
		"secondary_iso_images": [
			"{{ user `answer_media` }}"
		],

		"cpus": "{{ user `cpus` }}",
		"memory": "{{ user `memory` }}",
        "disk_size": "{{ user `disk_size`}}",
		"generation": 2,
		"enable_secure_boot": "true",

		"boot_wait": "0s",
		"boot_command": [
			"a<wait>a<wait>a"
		],
        "shutdown_command": "shutdown",

        "communicator": "winrm",
        "winrm_username": "Administrator",
        "winrm_password": "",
        "winrm_host": "",
        "winrm_port": 5985,
        "winrm_timeout": "60m",
        "winrm_insecure": "true",
        "winrm_use_ntlm": "false",
        "winrm_use_ssl": "true"
    }
    ]
} 
```

### Operating system and Environment details

Windows 2012 R2 with Hyper-V.

### Log Fragments and crash.log files

Unable to provide at this time, will try to update.

### PR DESCRIPTION
**Pull request by SwampDragons**
Adds much more comprehensive documentation for connecting to SSH and WinRM, as well as guides for preseeding ubuntu and windows. 

Closes #8150

### CODE REVIEW

File: website/source/docs/communicators/ssh.html.md [hunk 1]
```diff
@@ -1,41 +1,58 @@
...
+be performed by a (boot_command)[TODO LINK] that references a file which
+provides answers to the normally-interactive questions you get asked when
+installing an operating system. The name of this file varies by operating
+system; some common examples are the "preseed" file required by Debian, the
+"kickstart" file required by CentOS or the "answer file", also known as the
+Autounattend.xml file, required by Windows. For simplicity's sake, we'll refer
+to this file as the "preseed" file in the rest of the documentation.
+
+If you are unfamiliar with how to use a preseed file for automatic
+bootstrapping of an image, please either take a look at our [quick guides](TODO ADD LINK) to
```
Comment (azr) [inline website/source/docs/communicators/ssh.html.md]: Woops ! The link is missing

File: website/source/docs/communicators/ssh.html.md [hunk 2]
```diff
@@ -1,41 +1,58 @@
...
+
+If you are building from a cloud image (for example, building on Amazon), there
+is a good chance that your cloud provider has already preconfigured SSH on the
+image for you, meaning that all you have to do is configure the communicator in
+the Packer template.
+
+However, if you are building from a brand-new and unconfigured operating system
+image, you will almost always have to perform some extra work to configure SSH
+on the guest machine. For most operating system distributions, this work will
+be performed by a (boot_command)[TODO LINK] that references a file which
```
Comment (azr) [inline website/source/docs/communicators/ssh.html.md]: here too

**Comment by SwampDragons** [comment]
Thanks for catching those missing links!

**Comment by azr** [comment]
😄

**Pull request merged**